### PR TITLE
Indicate how Cargo locates the manifest

### PR DESCRIPTION
**Pull request by wmstack**
Currently the only place where that is documented is in an obscure flag called `--manifest-path`. The behaviour of Cargo in regards to locating the manifest should be in a more obvious place, and this seems to be right about where I added it.

### CODE REVIEW

**Review by weihanglo — COMMENTED** [review]
First, thanks for your contribution. It's glad to see hard works on doc improvement!

Your observation is correct. Cargo does search `Cargo.toml` in current working dir and upward if not exist. But speaking to workspaces it becomes a bit complicated. Cargo searches for root workspace manifest by either traversing parent dirs or look up the field [`package.workspace`](https://doc.rust-lang.org/cargo/reference/manifest.html#the-workspace-field).

I might also be confused seeing a sentence about how Cargo locates `Cargo.toml` there, as the paragraph is mainly talking about the format of `Cargo.toml` but not any individual command. I would try to expand the description of [`cargo locate-project`](https://doc.rust-lang.org/cargo/commands/cargo-locate-project.html) with the search logic, and refer to it in anywhere needed.

**Comment by wmstack** [comment]
Would it be alright to reference [`cargo locate-project`](https://doc.rust-lang.org/cargo/commands/cargo-locate-project.html) in the manifest? I'd imagine that it wouldn't occur to a normal user to look there, so there needs to be a reference, even a short one, that indicates how the manifest is located *somewhere* obvious. Or, which directory would you prefer it was referenced in?

**Comment by wmstack** [comment]
Just a general question. Is it the job of the maintainer or the contributor to handle it when pull requests diverge? Does that generally depend on the open-source project?

**Comment by weihanglo** [comment]
I would expect something like this, and then we can expand the detail on `cargo-locate-project`. My writing is rusty. Feel free to polish it :)

```
The `Cargo.toml` file for each package is called its *manifest*. It contains
the metadata that Cargo needs to compile a package. Most Cargo subcommands
start from  locating the manifest file first, and then perform subsequent
operations. See `cargo locate-project` chapter for more detail on how cargo
finds the manifest file.

The file is written in the [TOML] format. Every manifest file consists of the following sections:
```

> Just a general question. Whose job is it to resolve merge conflicts when pull requests diverge from the main branch? Is it the contributor or the maintainer? Does that generally depend on the particular open-source project?

I would say it's PR authors' responsibilities unless the conflict is too complicated. If in that case, maintainers and contributors collaborate to resolve it. And I feel like only when it is really a conflict should you update your branch to master.

**Comment by weihanglo** [comment]
Ping @wmstack. Just checking in to see if you are still interested in working on this, or if you had any questions.

**Comment by weihanglo** [comment]
Hi. Looks like you're still working on this PR. The PR looks good, but it is not much helpful inkt linking to `cargo locate-project` without updating doc there. Are you willing to add docs for `cargo locate-project`? Or do you have any other thought on this?

r? @weihanglo

**Comment by wmstack** [comment]
Hmm, I was under the impression that I had to run a shell script to fix failing unit tests. However, when I did that, it seems that I lost the documentation that I put there at https://github.com/rust-lang/cargo/pull/10770/commits/755bdc1b48511c67b577664951f1d544ce7c94bf

**Comment by weihanglo** [comment]
Sorry for not saying in clear. To generate docs for commands, you need to follow [this guide](https://github.com/rust-lang/cargo/tree/master/src/doc#building-the-man-pages). That is, edit `src/doc/man/cargo-locate-project.md` and run `src/doc/build-man.sh`.

We're almost there! Only a few things to do before merging: 

* Could you try adding some words in `cargo-locate-project` doc to tell the difference that other commands usually locate for workspace first by default, whereas `cargo locate-project` need the `--workspace` flag?
* Could you do a clean-up for commits, such like rebasing this PR onto master and keeping only meaningful commits?

Thank you!

**Comment by wmstack** [comment]
Every time I run the shell script, the content that I have written in `cargo-locate-project` dissappears...

File: src/doc/src/commands/cargo-locate-project.md
Comment (weihanglo) [inline src/doc/src/commands/cargo-locate-project.md:21]: I think it behaves like this.

```suggestion
locating the manifest of a workspace member. In most other commands, cargo locates the workspace
root by default, while in `cargo locate-project`, the above flag needs to be specified.
```

**Review by weihanglo — COMMENTED** [review]
> Every time I run the shell script, the content that I have written in `cargo-locate-project` dissappears...

You need to edit `src/doc/man/cargo-locate-project.md` instead and generate others.

File: src/doc/src/commands/cargo-locate-project.md
Comment (weihanglo) [reply src/doc/src/commands/cargo-locate-project.md]: Really Sorry. I was wrong by misunderstanding what `cargo locate-project` does 😞. They should behave the same. Could you help me remove things after “In most other commands…”? Thank you :)

**Comment by wmstack** [comment]
Okay I fixed the typos and squashed all the commits into 1.

**Review by weihanglo — COMMENTED** [review]
Thank you @wmstack. It's my fault but can you do me one more favor for this?
https://github.com/rust-lang/cargo/pull/10770#discussion_r981073253

**Comment by wmstack** [comment]
Okay, I removed the last comment.

**Review by weihanglo — APPROVED** [review]
Thank you so much!!

**Comment by weihanglo** [comment]
Oh no. We need to run `src/doc/build-man.sh` again.

**Comment by wmstack** [comment]
Man, I had to touch up on skills I hadn't used in ages! Like reflog and rebase and such. Thanks as well!

**Pull request merged**
